### PR TITLE
Support configurable decimal divide zero

### PR DESCRIPTION
**Pull request by shuai-xu**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
add support for allowing decimal divide zero

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

### CODE REVIEW

File: src/Core/Settings.h [hunk 1]
```diff
@@ -490,6 +490,7 @@ class IColumn;
     M(UInt64, low_cardinality_max_dictionary_size, 8192, "Maximum size (in rows) of shared global dictionary for LowCardinality type.", 0) \
     M(Bool, low_cardinality_use_single_dictionary_for_part, false, "LowCardinality type serialization setting. If is true, than will use additional keys when global dictionary overflows. Otherwise, will create several shared dictionaries.", 0) \
     M(Bool, decimal_check_overflow, true, "Check overflow of decimal arithmetic/comparison operations", 0) \
+    M(Bool, decimal_allow_divide_zero, false, "Whether allow divide zeor of decimal arithmetic operations", 0) \
```
Comment (alexey-milovidov) [inline src/Core/Settings.h:493]: Typo.

**Comment by alexey-milovidov** [comment]
Tests needed.

**Comment by Algunenano** [comment]
This will never work. Decimal division returns a decimal, which can't contain NULL.

In SQL you can change `decimal_a / decimal_b` with `decimal_a / nullIf(decimal_a, 0)` so that the type changes and you get NULL.

If you want to add a setting I'd suggest looking into adding it as a pass in the analyzer so that it adds this check (transform 0 to NULL) automatically for the denominator under a setting.

**Comment by shuai-xu** [comment]
> This will never work. Decimal division returns a decimal, which can't contain NULL.
> 
> In SQL you can change `decimal_a / decimal_b` with `decimal_a / nullIf(decimal_a, 0)` so that the type changes and you get NULL.
> 
> If you want to add a setting I'd suggest looking into adding it as a pass in the analyzer so that it adds this check (transform 0 to NULL) automatically for the denominator under a setting.

Yes, you are right， I will try another way.

**Pull request closed without merging**